### ISSUE TITLE
NVME issues - wrong partition names if nvme is used (missing p)

### ISSUE THREAD

**Issue by jTd7bPLFb** [psy0rz/alpinebox#4]
I tried to install using your script (thank you btw.) on an nvme device.

the exports for partitions are not usable for nvme

ls -la /dev/nvme*
nvme0n1    nvme0n1p1  nvme0n1p2  nvme0n1p3  nvme0n1p4

invocation "sh install /dev/nvme0n1"

the exports in your script
export INSTALL_DISK=$DISK
export INSTALL_EFI_DEV=$DISK""2
export INSTALL_SWAP_DEV=$DISK""3
export INSTALL_ZPOOL_DEV=$DISK""4
export INSTALL_ZPOOL=${INSTALL_ZPOOL:-rpool}

so basically, they become in this case 
/dev/nvme0p1
/dev/nvme0p12
/dev/nvme0p13
/dev/nvme0p14

which is wrong, as it misses the p

I noticed this on mkswap command failing.
mkswap: can't open '/dev/nvme0n13': No such file or directory

maybe put a if/then in the install.sh script, and also a small echo for each of the variables?

**Comment by jTd7bPLFb** [comment]
I tried once again with the image-install method.
Did not work for me.
1. boot alpine iso
2. setup-network, setup-dns, setup-ssh

using the simple install image method

localhost:~# sh image /dev/nvme0n1
ALPINEBOX: Getting latest stable version...
ALPINEBOX: Writing https://github.com/psy0rz/alpinebox/releases/latest/download/alpine.img.gz to disk /dev/nvme0n1 ...
Connecting to github.com (140.82.121.3:443)
Connecting to github.com (140.82.121.3:443)
Connecting to objects.githubusercontent.com (185.199.110.133:443)
writing to stdout
-                    100% |********************************|  467M  0:00:00 ETA
written to stdout
ALPINEBOX: All done, will reboot in 5 seconds..

after reboot, the machine hangs on 
'Tearing down USB controller'


4. so, I to using the other method, ,with the separate script, when using official alpine installer
5. just added the p to the disk/device names, as mentioned, and installer working without issues.


So, to sum up, I used the Alpine iso, that is why I use the non image install method.
:)

**Comment by psy0rz** [comment]
Thanks,

Are you sure it didn't just hang during the shutdown sequence? Did you try hard-resetting it to see if it would boot?

**Comment by jTd7bPLFb** [comment]
Yes.

**Comment by psy0rz** [comment]
Its hard to reproduce here. Could you make a screenshot/photo of it? I assume not since its already installed now :)

**Comment by jTd7bPLFb** [comment]
which one? the issue is as stated at the beginning

if disk is sda
partitions sda1, sda2, sda3... and so on

export INSTALL_EFI_DEV=$DISK""2
export INSTALL_SWAP_DEV=$DISK""3
export INSTALL_ZPOOL_DEV=$DISK""4

but when nvme is used
partions are prepended with p
nvme0p1, nvme0p2 ... and so on

so basically I would just put something like
grep nvme $DISK
if true then just add p

export INSTALL_EFI_DEV=$DISK"p"2
export INSTALL_SWAP_DEV=$DISK"p"3
export INSTALL_ZPOOL_DEV=$DISK"p"4

**Comment by psy0rz** [comment]
No i meant when you use the imaging method. Could you screenshot that at the point it fails?

**Comment by ben-auo** [comment]
FYI, this worked for my use case: https://github.com/ascension-association/zbm-void-ts/blob/main/zbm-void-ts.sh#L78

**Comment by neerdoc** [comment]
I'm a bit confused here. I thought I had the same issue, but for me it was not the install that didn't work, that actually worked fine. But when I added my second nvme disk to the root I got exactly this issue, that the 'p' is missing. I just added the 'p' in the add-disk script and was done with it, but definitely a nice fix to add some kind of detection of nvme disks.
